game-core/src/test/java/games/strategy/triplea/util/WrappedInvocationHandlerTest.java

```diff
@@ -19,23 +18,23 @@ public void equals_ShouldReturnTrueWhenOtherInstanceIsWrappedInvocationHandlerPr
     final Object proxy1 = newProxy(new WrappedInvocationHandler("test"));
     final Object proxy2 = newProxy(new WrappedInvocationHandler("test"));
 
-    assertTrue(proxy1.equals(proxy2));
+    assertEquals(proxy1, proxy2);
   }
 
   @Test
   public void equals_ShouldReturnFalseWhenOtherInstanceIsWrappedInvocationHandlerProxyWithUnequalDelegate() {
     final Object proxy1 = newProxy(new WrappedInvocationHandler("test1"));
     final Object proxy2 = newProxy(new WrappedInvocationHandler("test2"));
 
-    assertFalse(proxy1.equals(proxy2));
+    assertNotEquals(proxy1, proxy2);
   }
 
   @Test
   public void equals_ShouldReturnFalseWhenOtherInstanceIsProxyWithoutWrappedInvocationHandler() {
     final Object proxy1 = newProxy(new WrappedInvocationHandler("test"));
     final Object proxy2 = newProxy((proxy, method, args) -> null);
 
-    assertFalse(proxy1.equals(proxy2));
+    assertNotEquals(proxy1, proxy2);
   }
 
```

Comment:
@ssoloff Thoughts on this? Tagging you, because you used it probably more than others

Reply:
I don't think you can use EV here because `WrappedInvocationHandler` does not actually obey the `equals()`/`hashCode()` contract.  These are simply characterization tests that were added to capture the current behavior of this type before fixing a bug.

Also, using `assertFalse()` and `assertTrue()` here were conscious decisions because we're testing the return value of `equals()`, not necessarily whatever `assertNotEquals()` and `assertEquals()` are testing.  For all intents and purposes, the implementations of the latter are equivalent to the former except they handle `null`s.  However, the choice of `assertFalse()` and `assertTrue()` was to explicitly call out the test of the return value and potentially avoid any false positives (e.g. if `newProxy()` for some reason returned `null`).